/jds/bin/diff -uprN nautilus-2.26.3.old/libnautilus-private/nautilus-icon-container.c nautilus-2.26.3/libnautilus-private/nautilus-icon-container.c
--- nautilus-2.26.3.old/libnautilus-private/nautilus-icon-container.c	2009-05-18 20:44:26.000000000 +0100
+++ nautilus-2.26.3/libnautilus-private/nautilus-icon-container.c	2009-07-13 11:16:32.205356000 +0100
@@ -1841,6 +1841,12 @@ find_empty_location (NautilusIconContain
 	/* Start the icon on a grid location */
 	snap_position (container, icon, &start_x, &start_y);
 
+	while (start_x < 0)
+		start_x += SNAP_SIZE_X;
+
+	while (start_y < 0)
+		start_y += SNAP_SIZE_Y;
+
 	icon_position.x0 = start_x;
 	icon_position.y0 = start_y;
 	icon_position.x1 = icon_position.x0 + icon_width;
